### ISSUE TITLE
C function calls appear as "self"

### ISSUE THREAD

**Issue by untoreh** [joerick/pyinstrument#101]
When I have functions in the chunk of code im profiling like
```
some_class.some_method()
```

Pyinstrument will aggreggate all of them under `[self]` rendering them undiscernible.

**Comment by joerick** [comment]
Thanks! Looks like that's really because `arr.cumsum()` is a C function, not a Python function. I tested the code using a 'print profiler'

```python
import sys
import numpy as np

arr = np.random.randint(0, 10000, 10000)

def print_profiler(frame, event, arg):
    print(event, frame.f_code.co_name, frame.f_code.co_filename, arg)

sys.setprofile(print_profiler)

for i in range(2):
        arr.cumsum()

sys.setprofile(None)
```

 which outputs

```
c_call <module> examples/np_static.py <built-in method cumsum of numpy.ndarray object at 0x7fb7a8061120>
c_return <module> examples/np_static.py <built-in method cumsum of numpy.ndarray object at 0x7fb7a8061120>
c_call <module> examples/np_static.py <built-in method cumsum of numpy.ndarray object at 0x7fb7a8061120>
c_return <module> examples/np_static.py <built-in method cumsum of numpy.ndarray object at 0x7fb7a8061120>
c_call <module> examples/np_static.py <built-in function setprofile>
```

That indicates that it might be possible to track time spent in these functions. Let me look into it a little more...

**Comment by joerick** [comment]
#103 should fix this particular problem for you - if you have a chance to test it I'd be grateful!

**Comment by untoreh** [comment]
thanks, it works.
what about this? All three calls return similar stats, so what is `[self]`? The for loop, or the pyinst class itself?
```python
import numpy as np
from pyinstrument import Profiler
profiler = Profiler()

class Test:
    def run(self):
        None
    @staticmethod
    def run2():
        None

def run3():
    None

t = Test()

profiler.start()

for i in np.arange(100000):
        #t.run()
        #Test.run2()
        run3()

profiler.stop()
print(profiler.output_text(unicode=True, color=True))
```

**Comment by joerick** [comment]
Fixed in 3.2.0
